extensions/panache/hibernate-reactive-rest-data-panache/deployment/src/test/resources/import.sql

```diff
@@ -1,7 +1,7 @@
 insert into collection(id, name) values ('empty', 'empty collection');
 insert into collection(id, name) values ('full', 'full collection');
 
-insert into item(id, name, collection_id) values (nextval('hibernate_sequence'), 'first', 'full');
-insert into item(id, name, collection_id) values (nextval('hibernate_sequence'), 'second', 'full');
-
+insert into item(id, name, collection_id) values (1, 'first', 'full');
+insert into item(id, name, collection_id) values (2, 'second', 'full');
+alter sequence Item_SEQ restart with 3;
```

Comment:
So we don't need to do the `+ 50` here? /cc @yrodiere 

Asking because it might work with the current tests but might be a problem if we add more, and it will probably come as a surprise. Asking for all the sequence restarts in the PR.

Reply:
You don't need the +50 here - it's fine.

Reply:
Meaning HR doesn't support optimizers, or we don't `persist` in these tests, or we don't care about negative/conflicting IDs here, or you looked into these tests and we don't get negative IDs for some reason?

Reply:
We discussed this extensively on zulip the other day?

The problem of the "+50" only manifests if your sequence increment isn't matching what Hibernate ORM expects - essentially if you're running an ORM6 application with default sequence generators, on a schema which was created for ORM 5 (and not updated to match).

But since ORM generated the schema, it's going to increment correctly:
```
Hibernate: 
    create sequence EmptyListItem_SEQ start with 1 increment by 50
Hibernate: 
    create sequence Item_SEQ start with 1 increment by 50
```

If you test it, you'll see the next insert will have id=4.

Reply:
> If you test it, you'll see the next insert will have id=4.

My bad, should be id=3 I think

Reply:
> The problem of the "+50" only manifests if your sequence increment isn't matching what Hibernate ORM expects

That's not what I saw in my tests.

Quoting myself from [our discussion](https://hibernate.zulipchat.com/#narrow/stream/132094-hibernate-orm-dev/topic/Default.20increment.20size.20for.20sequences/near/339301201):

```quote
 Even if you update the increment size, if you don't initialize the sequence to the correct value, ORM 6 will start reusing old IDs.
```

> If you test it, you'll see the next insert will have id=4.

That's not consistent with the behavior of the `pooled` optimizer, which upon retrieving a sequence value of `3`, is supposed to use a pool of values going from `4-50+1=-45 to `4`. That's just how it's supposed to work.

The only ways I could see this working is if for some reason:

* Hibernate ORM deals with this problem in creative ways.
* Hibernate Reactive does not use the `pooled` optimizer.
* `alter sequence Item_SEQ restart with 3;` was actually setting the "previous" value of a sequence, so that the next increment actually returns `53`. That's not the behavior of PostgreSQL nor MariaDB as far as I can see.
* OR the sequence is incremented once before the first call to `persist` but the resulting value is never used.

Anyway, I guess neither of us has time to investigate more.

> But since ORM generated the schema, it's going to increment correctly:

The core of the problem here is not the schema, but the the next value of the sequence and how Hibernate ORM (Reactive) is going to interpret it.

Reply:
Yes you're right but it's using a different generator here. It's all setup explicitly on the model.

Reply:
so sorry, my previous comment was wrong as I actually had a different model in front of me. Ignore me :)